frontend/src/components/Sidepanel/Panels/TestingLab/TestingLab.tsx

```diff
@@ -242,8 +230,7 @@ const TestingLab = () => {
           <Button icon={<ChevronRight size={23} />}
             disabled={
               traceIdx >= lastTraceIdx ||
-              noInitialState ||
-              (projectType === 'TM' && !showTraceTape)
+              noInitialState || noFinalState || !pathToFinal
```

Comment:
would put these in like a `invalidMachine` variable or something so it can be easily referenced in the other places that need to check